### ISSUE TITLE
Load automatically is disabled.

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#3638]
##### Brief Description

OS: Windows x64
qTox version: v1.5.1-8-gfd04a28.
Commit hash:  fd04a2832aa885d260ded4e476eefcddd49237b3
toxcore:  0.0.0

Reproducible: Always
##### Steps to reproduce

Try to login automatically.
##### Observed Behavior

Load automatically is disabled.
##### Expected Behavior

Load automatically is enabled.
##### Additional Info

How to enable it?
I guessed that it can be related to encrypted profile.
If yes then some some questions:
What is encrypted profile? **How to decrypt it?**
If password used only for encryption of local profile then what should be secured as secret?
Does local profile contains private key (and it is secret)? Is public key is my ToxID?

**Comment by Diadlo** [comment]
Hello. The encrypted profile means, that you setup password to this profile.
If you are using the encrypted profile it can't be loaded automaticly.
You should store in secret your `.tox` file (it's contains your private key) and password if you have encrypted profile.
If I understand correct, ToxID is your public key.

**ADDED:** If you want automatically login you should:
1) Open profile info and click `Remove password`
2) Go to settings, `General` tab. And enable `Autostart`

**Comment by initramfs** [comment]
To expand on @Diadlo's point, once you set up a password, your tox profile and chat database are automatically encrypted by a key derived from your password. As such, it becomes impossible to automatically load a profile without the password (since we need to decrypt the profile before loading).

The way that encryption is performed varies between files (chat DB for example uses SQLite encryption whereas the tox profile uses functions from toxencryptsave). Decryption of data happens automatically when you enter your password to "login" to qTox.

As such, "automatic profile loading" is not a supported feature (it's impossible) for qTox profiles with a password. Unless this issue refers to non-encrypted profiles (i.e. profiles with no password) cannot be automatically loaded, this is not a bug.

Please modify your issue to include whether you are trying to load a profile **with** or **without** a password.

**Comment by ghost** [comment]
Hello...
Thanks for quick response... now all is clear... ))

Maybe I can suggest the following:
1) Move profile UI to new tab (available by settings button). It can be first tab there and have title "Profile".
2) Click on Profile will open this tab in settings.

It can be more clear for user.. it is intuitive to get all settings in one place. And also such behavior will solve the case when user expects profile settings by username click.

**Comment by initramfs** [comment]
@maxim-boligatov I take your response as you've tried to use automatic profile loading with an encrypted profile and thus will be closing your issue.

Regarding the UI changes you suggested, qTox currently has a few UI changes being developed, most notably #3631 and #3457. You may be interested in looking at the changes proposed by those pull requests to see if they incoperate something along the lines of what you want. If not, feel free to open a **new** issue describing in detail the proposed changes (proposing it as part of this issue will limit it's visibility) and someone may implement this in the future.

And for the record, I think that's a decent suggestion :)

**Comment by ghost** [comment]
> Please modify your issue to include whether you are trying to load a profile with or without a password.

I think we can close this issue... it will be informative for others users.
Also I think tooltip can be added on disabled "Load automatically" control.
So user will get at least minimal information how it works and why it is disabled.
It can be very helpful for new users.

**Comment by zetok** [comment]
> Also I think tooltip can be added on disabled "Load automatically" control.

There is one.

> Maybe I can suggest the following:
> 1) Move profile UI to new tab (available by settings button). It can be first tab there and have title "Profile".
> 2) Click on Profile will open this tab in settings.
> 
> It can be more clear for user.. it is intuitive to get all settings in one place. And also such behavior will solve the case when user expects profile settings by username click.

No, having _profile_ in _settings_ is not more intuitive.
If you want to open a new issue about it, please relate to discussion in #1012 and 1328.

**Comment by ghost** [comment]
> _Also I think tooltip can be added on disabled "Load automatically" control._      
> 
> There is one.    

It is absent at least on:

> OS: Windows x64
> qTox version: v1.5.1-8-gfd04a28.
> Commit hash: fd04a28
> toxcore: 0.0.0

Maybe some nightly build already solves it, but current stable version works as I described above.

> No, having profile in settings is not more intuitive.

I cannot agree...
To be more clear.... Imagine situation when you need 2 accounts inside single instance (it is not really needed in such way but just for analyzing).

So to setup each account we will need provide at least own settings for:
Autoaway
Autoaccept
Download folder
Maybe some chat settings and emoticons
Privacy
as well as ToxId from profile!

All above is some sort of settings which should be set up to make **tox account working**.
In other words what is difference between ToxId and NoSpam seed!? All of them belongs to profile identification.

I am sure it is best way to allow click on Profile and go to first tab in settings named Profile.    

## 

This is just a response for your last message. But of course we can create new issue and continue this conversation with all community. But I am new for qTox and you already have some plans for redesign. So I am not sure if it is still actual...

**Comment by Mikaela** [comment]
I installed qtox 1.5.2-1 yesterday on Arch Linux and when I opened it, I was asked for username and password. I wasn't told that if I give a password (which appeared required), I will be unable to load the profile automatically.

I think the tooltip also doesn't work so well as I discovered it exists only after finding this issue. Could it be moved as text under the checkbox and above the buttons?

**Comment by zetok** [comment]
@Mikaela 

> I wasn't told that if I give a password (which appeared required), I will be unable to load the profile automatically.

That's a feature.

> I think the tooltip also doesn't work so well as I discovered it exists only after finding this issue. Could it be moved as text under the checkbox and above the buttons?

Since b5efc2bb7124ccbbd256792c524ffffdd02fab3f the checkbox is hidden if profile is password-protected.

Added a FAQ entry that explains it: https://github.com/qTox/qTox/wiki/FAQ#can-a-profile-be-automatically-loaded-if-its-password-protected

**Comment by Mikaela** [comment]
> That's a feature.

But I don't trust myself with passwords so I generated a random one that I won't remmeber making me need to check the password every time I started qTox or booted laptop increasing the risks of forgetting password database open or other human error that endangers more than one Tox profile.

> Added a FAQ entry that explains it: https://github.com/qTox/qTox/wiki/FAQ#can-a-profile-be-automatically-loaded-if-its-password-protected

Is the FAQ or other documentation linked anywhere clearly from the app or https://tox.chat/ ?

**Comment by zetok** [comment]
> Is the FAQ or other documentation linked anywhere clearly from the app or https://tox.chat/ ?

User manual is linked to from qTox website: https://qtox.github.io/
Once FAQ will be good enough I plan on adding a link to it on the website.
